tck/methods/account.go

```diff
@@ -0,0 +1,279 @@
+		AddHbarTransfer(alias, hedera.NewHbar(float64(fromAliasParams.InitialBalance))).
+		Execute(a.sdkService.Client)
+
+	if err != nil {
+		fmt.Println(err)
+		return nil, jrpc2.Errorf(-32603, "Internal error").WithData(ErrorStatus{Status: err.Error()})
+	}
+	receipt, err := resp.GetReceipt(a.sdkService.Client)
+	var accId string
+	if receipt.Status == hedera.StatusSuccess {
+		accId = receipt.AccountID.String()
+	}
+
+	return &AccountResponse{AccountId: accId, Status: receipt.Status.String()}, nil
+}
+
+// GetAccountInfo Get info for a given accountId and return a custom object containing aggregated information about the account
+func (a *AccountService) GetAccountInfo(_ context.Context, accountId string) (*AccountInfoResponse, error) {
+	account, _ := hedera.AccountIDFromString(accountId)
+
+	resp, err := hedera.NewAccountInfoQuery().SetAccountID(account).SetGrpcDeadline(&threeSec).Execute(a.sdkService.Client)
+	if err != nil {
+		return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
+	}
+
+	return &AccountInfoResponse{AccountID: resp.AccountID.String(),
+		Balance:                       resp.Balance.String(),
+		Key:                           resp.Key.String(),
+		AccountMemo:                   resp.AccountMemo,
+		MaxAutomaticTokenAssociations: resp.MaxAutomaticTokenAssociations,
+		AutoRenewPeriod:               resp.AutoRenewPeriod}, nil
+}
+
+type UpdateAccountParams struct {
+	AccountId string `json:"accountId"`
+	// Params for updateAccountKey
+	NewPublicKey  string `json:"newPublicKey"`
+	OldPrivateKey string `json:"oldPrivateKey"`
+	NewPrivateKey string `json:"newPrivateKey"`
+	// Params for updateAccountMemo
+	Key  string `json:"key"`
+	Memo string `json:"memo"`
+}
+
+// UpdateAccount transaction to be associated with hedera.AccountUpdateTransaction and to be enough generic, so different
+// updates can be calling this func
+func (a *AccountService) UpdateAccount(_ context.Context, params UpdateAccountParams) (*AccountResponse, error) {
+	if params.NewPrivateKey != "" { // Proceed with UpdateAccountMemo
+		return a.UpdateAccountKey(params.AccountId, params.NewPublicKey, params.OldPrivateKey, params.NewPrivateKey)
+	} else if params.Memo != "" { // Proceed with UpdateAccountKey
+		return a.UpdateAccountMemo(params.AccountId, params.Key, params.Memo)
+	}
+	return nil, nil
+}
+
+// DeleteAccount deletes a provided account by signing the transaction with the key of that account
+func (a *AccountService) DeleteAccount(_ context.Context, params DeleteAccountParams) (*AccountResponse, error) {
+	accId, _ := hedera.AccountIDFromString(params.AccountId)
+	recipientId, _ := hedera.AccountIDFromString(params.RecipientId)
+	tx, err := hedera.NewAccountDeleteTransaction().SetGrpcDeadline(&threeSec).SetAccountID(accId).SetTransferAccountID(recipientId).FreezeWith(a.sdkService.Client)
+	if err != nil {
+		return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
```

Comment:
Would guess that `Internal error` will not be the only received error. Should standardise a bit more around error definitions in the TCK projects.